src/Renderer/NavigatorRenderer.cpp

```diff
@@ -0,0 +1,378 @@
+  polyline_frame[6].x = polyline_frame[0].x; //
+  polyline_frame[6].y = polyline_frame[5].y; //
+  polyline_frame[7].x = top_left.x + 2 / 20.0 * rc_height ; //
+  polyline_frame[7].y = polyline_frame[4].y; //
+  polyline_frame[8].x = top_left.x; //
+  polyline_frame[8].y = polyline_frame[3].y;//
+  polyline_frame[9].x = polyline_frame[7].x; //
+  polyline_frame[9].y = polyline_frame[2].y; //
+  polyline_frame[10].x = polyline_frame[0].x; //
+  polyline_frame[10].y = polyline_frame[0].y; //
+  
+  canvas.Select(look.background_pen);
+  canvas.Select(look.background_brush);
+  canvas.DrawPolygon( polyline_frame, 11);
+
+  canvas.Select(look.frame_pen);
+  canvas.Select(look.frame_brush);
+  canvas.DrawPolyline( polyline_frame, 11);
+}
+
+
+void 
+NavigatorRenderer::DrawText(Canvas &canvas, const Waypoint &wp_current, 
+                                const PixelRect &rc, const NavigatorLook &look) {
+
+  const auto &basic = CommonInterface::Basic();
+  const auto &calculated = CommonInterface::Calculated();
+  TCHAR current_speed[256] = "---";
+
+  if (basic.ground_speed_available)
+    FormatSpeed(current_speed, Units::ToSysUnit(basic.ground_speed, Unit::METER_PER_SECOND),
+              Unit::KILOMETER_PER_HOUR, true, true);
+  
+  auto current_speed_s = static_cast<std::string>(current_speed);
+    
+  TextRenderer text_renderer1;
+  text_renderer1.SetVCenter(true);
+  text_renderer1.SetCenter(false);
+  text_renderer1.SetControl(true);
+
+  TextRenderer text_renderer2;
+  text_renderer2.SetVCenter(true);
+  text_renderer2.SetCenter(true);
+  text_renderer2.SetControl(true);
+  
+  canvas.Select(*look.font);
+
+  bool has_started = calculated.ordered_task_stats.start.task_started;
+
+  const auto time_elapsed = TimeStamp{FloatDuration{calculated.ordered_task_stats.total.time_elapsed}};
+  const RoughTimeDelta r{};
+
+  auto time_elapsed_s = FormatLocalTimeHHMM(time_elapsed, r).c_str();
+  has_started ? time_elapsed_s : time_elapsed_s = "--:--";
+
+  const auto time_start = calculated.ordered_task_stats.start.time;
+  auto time_start_s = FormatLocalTimeHHMM(time_start,
+                                  CommonInterface::GetComputerSettings().utc_offset).c_str();
+  has_started ? time_start_s : time_start_s = "--:--";
+
+  const auto time_local = FormatLocalTimeHHMM(basic.time,
+                                  CommonInterface::GetComputerSettings().utc_offset).c_str();
+
+  const auto time_planned = TimeStamp{FloatDuration{calculated.ordered_task_stats.total.time_planned}};
+  auto time_planned_s = FormatLocalTimeHHMM(time_planned, r).c_str();
+  has_started ? time_planned_s : time_planned_s = "--:--";
+
+  const auto arrival_planned = TimeStamp{FloatDuration{time_start.ToDuration() + time_planned.ToDuration()}};
+  auto arrival_planned_s = FormatLocalTimeHHMM(arrival_planned, 
+                                  CommonInterface::GetComputerSettings().utc_offset).c_str(); 
+  has_started ? arrival_planned_s : arrival_planned_s = "--:--";
```

Comment:
Funny indentation all over.

Reply:
Ok, updated. hope it's better now. 
useful standart tool under VScodium to see predicate... but it could be tricky for the indentation